processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -103,14 +101,22 @@ private void updateState(SubscriptionStateListener.State newState) {
                      .collect(Collectors.toSet());
     }
 
-    private void commitCompletedOffsets(Consumer<?, ?> consumer) {
+    private void commitCompletedOffsets(Consumer<?, ?> consumer, boolean sync) {
         Map<TopicPartition, OffsetAndMetadata> commitOffsets = contexts.commitOffsets();
         if (commitOffsets.isEmpty()) {
-            logger.debug("No new offsets to commit, skipping commit");
+            log.debug("No new offsets to commit, skipping commit");
             return;
         }
-        logger.debug("Committing offsets: {}", commitOffsets);
-        consumer.commitSync(commitOffsets);
+        log.debug("Committing offsets {}: {}", sync ? "SYNC" : "ASYNC", commitOffsets);
+        if (sync) {
+            consumer.commitSync(commitOffsets);
+        } else {
+            consumer.commitAsync(commitOffsets, (offsets, exception) -> {
+                if (exception != null) {
+                    log.warn("Offset commit failed asynchronously", exception);
```

Comment:
is it worth logging the offsets?

Reply:
`offsets.toString()` is really a debug string with unpredictable length (proportional to number of partitions) and I don't think it's good idea to put it in level >= warn which is likely enabled in most of applications.
Also,
* Even if we get offset values that are failed to commit, I don't think there's almost nothing we can do with it
* With much lower logging level (debug) it's written once in above debug logging

any thoughts?

Reply:
I see. I guess we can get enough clues from the exception itself!